buildpack.md

```diff
@@ -174,19 +180,22 @@ The order of the buildpacks in the group MUST otherwise be preserved.
 
 The `/bin/detect` executable in each buildpack, when executed:
 
-1. MAY examine the app directory and environment variables.
-2. MAY emit error, warning, or debug messages to `stderr`.
-3. MAY receive a TOML-formatted map called a Build Plan on `stdin`.
-4. MAY output changes to the Build Plan on `stdout`.
-5. MUST set an exit status code as described in the [Buildpack Interface](#buildpack-interface) section.
+- MAY read the app directory.
+- MAY read the detect environment as defined in the [Environment](#environment) section.
+- MAY emit error, warning, or debug messages to `stderr`.
+- MAY receive a TOML-formatted [Build Plan](#build-plan-(toml)) on `stdin`.
+- MAY contribute to the Build Plan by writing zero or more files to `<platform>/plan/`.
+- MUST set an exit status code as described in the [Buildpack Interface](#buildpack-interface) section.
+
+For each `/bin/detect`, the Build Plan received on `stdin` MUST be a map derived from the combined Build Plan contributions of all previous `/bin/detect` executables.
+In order to make an individual contribution to the Build Plan, a `/bin/detect` executable MUST write a file to `<platform>/plan/<name>` such that `<name>` is the desired top-level key and the file contents are the desired top-level value.
 
-For each `/bin/detect`, the Build Plan received on `stdin` MUST be a combined map derived from the output of all previous `/bin/detect` executables.
 The lifecycle MUST construct this map such that the top-level values from later buildpacks override the entire top-level values from earlier buildpacks.
-The lifecycle MUST NOT include any changes in this map that are output by optional buildpacks that returned non-zero exit statuses.
-The final Build Plan is the complete combined map that includes the output of the final `/bin/detect` executable.
+The lifecycle MUST NOT include any changes in this map that are contributed by optional buildpacks that returned non-zero exit statuses.
```

Comment:
This describes a well-behaved buildpack, but is this also going to be enforced by the lifecycle?

Reply:
This is already enforced by the lifecycle. 😄 

The phrasing ("lifecycle MUST NOT include any changes") suggests this to me, but happy to clarify further if needed.

Reply:
That's good enough for me.